### PR TITLE
Mm def

### PR DESCRIPTION
**Pull request by niruiyu**
# Description

The PR adds all interfaces used in x86 standalone MM design.

- [ ] Breaking change?
  - **Breaking change** - Will this cause a break in build or boot behavior?
  - Examples: Add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [x] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

X86 standalone MM implementation is implemented based on the interfaces defined in this PR.

## Integration Instructions

NA

### CODE REVIEW

**Comment by leiflindholm** [comment]
Please add a proper PR description, as you would a cover letter if we were reviewing on the mailing list.

File: MdeModulePkg/Include/Guid/MmCommBuffer.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#ifndef MM_COMM_BUFFER_H_
+#define MM_COMM_BUFFER_H_
+
+///
+/// The GUID of the MM Communication buffer HOB.
+///
+#define MM_COMM_BUFFER_HOB_GUID \
+  { 0x6c2a2520, 0x0131, 0x4aee, { 0xa7, 0x50, 0xcc, 0x38, 0x4a, 0xac, 0xe8, 0xc6 }}
+
+
```
Comment (jiaxinwu) [inline MdeModulePkg/Include/Guid/MmCommBuffer.h:19]: Remove one blank line to pass the uncrustify check

File: StandaloneMmPkg/StandaloneMmPkg.dec [hunk 2]
```diff
@@ -29,6 +29,9 @@
   ##                 MM Memory Operation.
   MemLib|Include/Library/StandaloneMmMemLib.h
 
+[LibraryClasses.X64.PEIM]
```
Comment (jiaxinwu) [inline StandaloneMmPkg/StandaloneMmPkg.dec:32]: add intel copyright

**Comment by jiaxinwu** [comment]
Others are good to me.

**Comment by jiaxinwu** [comment]
please remove one blank line to pass the uncrustify check

File: StandaloneMmPkg/Include/Library/MmPlatformHobProducerLib.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  * Single EFI_HOB_MEMORY_ALLOCATION_MODULE to describe the MM region used by MM Core.
+  * Multiple EFI_HOB_RESOURCE_DESCRIPTOR to describe the non-MM regions and their access permissions.
+    Note: All accessible non-MM regions should be described by EFI_HOB_RESOURCE_DESCRIPTOR HOBs.
+  * Single GUIDed (gMmCommBufferHobGuid) HOB to identify MM Communication buffer in non-MM region.
+  * Multiple GUIDed (gSmmBaseHobGuid) HOB to describe the SMM base address of each processor.
+  * Multiple GUIDed (gMpInformation2HobGuid) HOB to describe the MP information.
+  * Single GUIDed (gMmCpuSyncConfigHobGuid) HOB to describe how BSP synchronizes with APs in x86 SMM.
+  * Single GUIDed (gMmAcpiS3EnableHobGuid) HOB to describe the ACPI S3 enable status.
+  * Single GUIDed (gEfiAcpiVariableGuid) HOB to identify the S3 data root region in x86.
+	* Single GUIDed (gMmProfileDataHobGuid) HOB to describe the MM profile data region.
```
Comment (td36) [inline StandaloneMmPkg/Include/Library/MmPlatformHobProducerLib.h:32]: The line 32 starts with a TAB. Need to replace the TAB with blank space. Or CI will fail with:
'TAB' character is not allowed in source code, please replace each 'TAB' with two spaces.

**Pull request closed without merging**